Source/Extensions/URLRequest.swift

```diff
@@ -0,0 +1,27 @@
+import Foundation
+
+public extension URLRequest {
+  var jsonPayload: Any {
+    get {
+      return httpBody.flatMap { try? JSONSerialization.jsonObject(with: $0) } ?? [:]
+    }
+    set {
+      httpBody = try? JSONSerialization.data(withJSONObject: newValue)
+    }
+  }
+
+  @available(macOSApplicationExtension 10.10, *)
+  var formURLEncodedPayload: [URLQueryItem] {
+    get {
+      var components = URLComponents()
+      components.query = httpBody.flatMap { String(data: $0, encoding: .utf8) }
+      return components.queryItems ?? []
+    }
+
+    set {
+      var components = URLComponents()
+      components.queryItems = newValue
+      httpBody = components.query!.data(using: .utf8)
```

Comment:
Unless I'm mistaken, I believe that setting `queryItems` should always update `query`. My assumption is that if you can construct a `URLQueryItem`, you can always get a valid query string. The return type of this expression is still optional, but then maybe it would be fine to just be more conservative and use optional chaining here?